src/core/geometry/qgssphere.h

```diff
@@ -0,0 +1,188 @@
+     *
+     * \see centerX()
+     * \see centerY()
+     * \see centerZ()
+     * \see setCenter()
+     */
+    QgsPoint center() const SIP_HOLDGIL;
+
+    /**
+     * Returns the x-coordinate of the center of the sphere.
+     *
+     * \see center()
+     * \see centerY()
+     * \see centerZ()
+     * \see setCenter()
+     */
+    double centerX() const { return mCenterX; }
+
+    /**
+     * Returns the y-coordinate of the center of the sphere.
+     *
+     * \see center()
+     * \see centerX()
+     * \see centerZ()
+     * \see setCenter()
+     */
+    double centerY() const { return mCenterY; }
+
+    /**
+     * Returns the z-coordinate of the center of the sphere.
+     *
+     * \see center()
+     * \see centerX()
+     * \see centerY()
+     * \see setCenter()
+     */
+    double centerZ() const { return mCenterZ; }
+
+    /**
+     * Sets the center point of the sphere.
+     * \see center()
+     */
+    void setCenter( const QgsPoint &center ) SIP_HOLDGIL;
+
+    /**
+     * Sets the center point of the sphere to (\a x, \a y, \a z).
+     * \see center()
+     */
+    void setCenter( double x, double y, double z ) SIP_HOLDGIL { mCenterX = x; mCenterY = y; mCenterZ = z; }
+
+    /**
+     * Returns the radius of the sphere.
+     *
+     * \see setRadius()
+     * \see diameter()
+     */
+    double radius() const SIP_HOLDGIL { return mRadius; }
+
+    /**
+     * Sets the \a radius of the sphere.
+     *
+     * \see radius()
+     */
+    void setRadius( double radius ) SIP_HOLDGIL{ mRadius = radius; }
+
+    /**
+     * Returns the diameter of the sphere.
+     *
+     * \see radius()
+     */
+    double diameter() const SIP_HOLDGIL { return mRadius * 2; }
+
+    /**
+     * Returns the volume of the sphere.
+     */
+    double volume() const SIP_HOLDGIL;
+
+    /**
+     * Returns the surface area of the sphere.
+     */
+    double surfaceArea() const SIP_HOLDGIL;
+
+    /**
+     * Converts the sphere to a 2-dimensional circle.
+     */
+    QgsCircle toCircle() const SIP_HOLDGIL;
+
+    /**
+     * Returns the 3-dimensional bounding box containing the sphere.
+     */
+    QgsBox3d boundingBox() const SIP_HOLDGIL;
+
+#ifdef SIP_RUN
+    SIP_PYOBJECT __repr__();
+    % MethodCode
+    QString str;
+    if ( sipCpp->isNull() )
+    {
+      str = QStringLiteral( "<QgsSphere: null>" ).arg( sipCpp->centerX() ).arg( sipCpp->centerY() ).arg( sipCpp->centerZ() ).arg( sipCpp->radius() );
+    }
+    else
+    {
+      str = QStringLiteral( "<QgsSphere: (%1, %2, %3) radius %4>" ).arg( sipCpp->centerX() ).arg( sipCpp->centerY() ).arg( sipCpp->centerZ() ).arg( sipCpp->radius() );
+    }
+    sipRes = PyUnicode_FromString( str.toUtf8().constData() );
+    % End
+#endif
+
+  private:
+    double mCenterX = std::numeric_limits< double >::quiet_NaN();
+    double mCenterY = std::numeric_limits< double >::quiet_NaN();
+    double mCenterZ = std::numeric_limits< double >::quiet_NaN();
+    double mRadius = 1;
```

Comment:
Would it make sense to set the default radius to 0 so that QgsSphere() gives an object which is both null and empty?

Reply:
Done